Storage/StorageTests/Tools/GeneralAppSettingsStorageTests.swift

```diff
@@ -0,0 +1,179 @@
+import XCTest
+import Storage
+
+class GeneralAppSettingsStorageTests: XCTestCase {
+    private var fileStorage: MockInMemoryStorage!
+    private var storage: GeneralAppSettingsStorage!
+
+    override func setUp() {
+        fileStorage = MockInMemoryStorage()
+        storage = GeneralAppSettingsStorage(fileStorage: fileStorage)
+    }
+
+    override func tearDown() {
+        storage = nil
+        fileStorage = nil
+    }
+
+    func test_default_settings_when_settings_file_does_not_exist() {
+        // Given
+
+        // Make sure settings file doesn't exist
+        XCTAssertTrue(fileStorage.data.isEmpty)
+
+        // When
+        let settings = storage.settings
+
+        // Then
+        XCTAssertEqual(settings, GeneralAppSettings.default)
+    }
+
+    func test_save_settings_saves_to_file() throws {
+        // Given
+
+        // Make sure settings file doesn't exist
+        XCTAssertTrue(fileStorage.data.isEmpty)
+
+        // When
+        var settings = GeneralAppSettings.default
+        settings.installationDate = .init()
+        try storage.saveSettings(settings)
+
+        // Then
+        XCTAssertTrue(fileStorage.dataWriteIsHit)
+        XCTAssertEqual(storage.settings, settings)
+    }
+
+    func test_settings_publisher_publishes_values() throws {
+        // Given
+        var receivedSettings = [GeneralAppSettings]()
+        let cancelable = storage.settingsPublisher.sink { settings in
+            receivedSettings.append(settings)
+        }
+
+        let settings1 = GeneralAppSettings.default
+        var settings2 = settings1
+        settings2.knownCardReaders = ["READER2"]
+
+        // When
+        try storage.saveSettings(settings1)
+        try storage.saveSettings(settings2)
+
+        // Then
+        XCTAssertEqual(receivedSettings.count, 2)
+        XCTAssertEqual(receivedSettings, [settings1, settings2])
+
+        // Tear down
+        cancelable.cancel()
```

Comment:
🤦🏽‍♂️ True, I guess I hadn't dealt with cancelables for a while

Reply:
Actually, we need to hold on to the cancellable or it gets deallocated immediately and the subscription is canceled. So technically we don't need to call `cancel`, but if we don't, then the `cancelable` variable is unused and Swift throws a warning.

We could have a `Set<AnyCancellable>` property that we empty on `tearDown`, but since this is a test class, maybe it's ok to keep the redundant `cancel`?

Reply:
Yeah, it makes sense to me 👍